### PR TITLE
Add release helper

### LINKED ISSUE #1298: Simplify releases
We should be able to simplify releases to a short, one step process.

* Require the use a RELEASE_NOTES tag in every merge commit
* Create a shell script the requires GITHUB_TOKEN and possibly VERSION input.
* Parse every commit since the last release extracting RELEASE_NOTES, adding that to our CHANGELOG
* Run all tests and a goreleaser dry run
* Publish the release, reading the GitHub release notes from our changelog

This is blocked by #1296

### PR DESCRIPTION
**Pull request by dominikschulz**
This PR adds a small release helper along with some documentation.
This should allow anyone with sufficient permissions to cut and push new
releases.

Fixes #1298 

Signed-off-by: Dominik Schulz <dominik.schulz@gauner.org>

### CODE REVIEW

File: docs/releases.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  * If we're shipping possibly disruptive changes we bump the minor component
+  * So far we've never bumped the major component
+* Grep all `RELEASE_NOTES` from the commit messages and prepend a new section to the [CHANGELOG](../CHANGELOG.md)
+  * Major changes should be detailed in a few sentences
+* Write the new version to the VERSION file
+* Write the new version to the version.go file
+* Commit these changes to a branch and open a pull request
+* Once the PR has been merged immediately create a new tag `vX.Y.Z` and push it to GitHub
+  * This will kick-off the goreleaser GitHub Action to build and push the release
+  * If GHA is unavailable run `goreleaser` (with a valid `GITHUB_TOKEN` in your env) locally
```
Comment (AnomalRoil) [inline docs/releases.md:41]: Isn't that rather  `goreleaser release --release-notes <(go run helpers/changelog/main.go)`

File: helpers/release/main.go [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+	}
+	fmt.Fprintln(fh)
+
+	ofh, err := os.Open("CHANGELOG.md")
+	if err != nil {
+		return err
+	}
+	defer ofh.Close()
+
+	if _, err := io.Copy(fh, ofh); err != nil {
```
Comment (AnomalRoil) [inline helpers/release/main.go:211]: So you copy it, okay.

File: helpers/release/main.go [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+		return err
+	}
+
+	return os.Rename("CHANGELOG.new", "CHANGELOG.md")
```
Comment (AnomalRoil) [inline helpers/release/main.go:215]: And yet you rename it, which overwrite it? Why bother with the Copy then?

**Review by AnomalRoil — DISMISSED** [review]
Nice, I like it 👍 

BTW, wouldn't it be best to increment the version again to a `vX.Y.(Z+1)-rc1` directly after having tagged the release vX.Y.Z, so that the dev version is a RC and not a full version?

File: helpers/release/main.go (hunk 3 above)
Comment (dominikschulz) [reply helpers/release/main.go:215]: Because I need to prepend the new entries to the file.

**Comment by dominikschulz** [comment]
Yes, increasing the version right after the release would probably be a good idea.

**Review by AnomalRoil — APPROVED** [review]

**Pull request merged**